monkey/monkey_island/cc/repository/mongo_agent_repository.py

```diff
@@ -58,6 +58,17 @@ def get_running_agents(self) -> Sequence[Agent]:
         except Exception as err:
             raise RetrievalError(f"Error retrieving running agents: {err}")
 
+    def get_progenitor(self, descendant: Agent) -> Agent:
+        try:
+            parent_descendant = self.get_agent_by_id(descendant.parent_id)
+        except (RetrievalError, UnknownRecordError) as err:
+            raise err
```

Comment:
This try/except does nothing.
```suggestion
        parent_descendant = self.get_agent_by_id(descendant.parent_id)
```

Reply:
"parent_descendant" is a weird name. Those words are basically opposites. I think just `parent` would be good.